### DISCUSSION TITLE
Using route attribute inside a struct implementation

### DISCUSSION THREAD

**Discussion by saifsre** [rwf2/Rocket#2207] [General]
Hi, I wanted to ask if one is able to add a route attribute inside a struct implementation like this:

I am getting the error: struct is not supported in `trait`s or `impl`s
 
````
pub struct A;

impl A {
    #[get("/index")]
    pub fn index() -> &'static str {
        "\
        USAGE
        POST api/v1/
            accepts raw data in teh body of the request and responds with a URL of
            a page containing the body's content
        GET api/v1/<id>
            retrieves the content for the paste with id `<id>`
        "
    }
  }

**Comment by SergioBenitez** [comment]
At the moment you cannot, no. However there would be no benefit to doing this. You _can_ use a module to group routes, however, which gives you nearly the same functionality:

```rust
mod foo {
    #[get("/")]
    fn f() { }
}
```
